### PR TITLE
WX3 stabilization

### LINKED ISSUE #875: migrate to wxpython3.0
Dear Tribler developers,

In Debian, currently wxwidgets2.8 is dropping for jessie release.
So a migration from 2.8 to 3.0 is needed.
It is described in https://bugs.debian.org/759044 

Olly Betts made a patch for this. But there's still some issue. Please see the attachment in Debian bug report. It is named tribler-6.3.1+git20140902.9a5a2604-1.1-nmu.diff
Thanks a lot.

Yours Sincerely,
Paul

### LINKED ISSUE #1540: KeyError
I was using the down arrow key to preload everything in a channel, and this is what showed up in the log.
```
Traceback (most recent call last):
  File "/home/iron/tmp/tribler/Tribler/Main/vwxGUI/list_body.py", line 1396, in <lambda>
    self.Bind(wx.EVT_MENU, lambda event: self.SelectNextItem(True), id=adownId)
  File "/home/iron/tmp/tribler/Tribler/Main/vwxGUI/__init__.py", line 119, in invoke_func
    return func(*args, **kwargs)
  File "/home/iron/tmp/tribler/Tribler/Main/vwxGUI/list_body.py", line 1301, in SelectNextItem
    select].GetSize()[1] + 1) / self.GetScrollPixelsPerUnit()[1]
KeyError: 'K\r`\xc0p\xd69.\xb7\x98\xb7^\x01\xb8\xa4\xacS\xc7^\x8d'
```

Edited by @whirm: One copy of the stacktrace is enough.

### LINKED ISSUE #1560: file size search limiter render error
I could not upload the window image, so I had to post it on imgur, sorry in advanced.

https://i.imgur.com/SgCa4kc.png

### LINKED ISSUE #1585: The bittorrent icon on the download items should go
As we only support bittorrent transfers now.

### LINKED ISSUE #1704: Download table columns visibility editor has misleading UI and can't be opened without any downloads.
In tribler v5.4.3 downloads showed "ratio" field (and was able to sort by it).

Now I don't see it.

Without ratio or accumulated upload size user don't see the relative popularity of his shared content.

### LINKED ISSUE #1770: improve "Duplicate download" warning
Currently (v6.4.3) that warning just has an OK option that closes the whole `Add external .torrent` dialogue. I think it could be improved by:

a) automatically switching the main window to the Downloads section and selecting the torrent in question, or
b) adding a `select different .torrent` button, so one can start another torrent faster.

Thanks for your work on Tribler :-)

### LINKED ISSUE #1941: v6.5.0-rc7: "Connections: <empty>"
Looks like when connections number is large enough (now downloading tails-i386-2.0) it doesn't fit and gets wrapped around.

Now I see this

    Connections:
    197

(with line break)

### LINKED ISSUE #1947: Color of text turns black on OS X after a while
After starting Tribler, the text color of the labels in Tribler turns black. This happens on the devel branch.

### LINKED ISSUE #2048: List items cannot be selected very well on OS X using wx3
In the wx3 GUI on OS X, list items are not always selectable. I suspect that the label is eating the click event instead of the list row.

### LINKED ISSUE #2050: Key error in list_body
When pressing the down key when not having selected anything you get the following stacktrace:

Traceback (most recent call last):
  File "/home/laurens/Documents/thesis/tribler/Tribler/Main/vwxGUI/list_body.py", line 1399, in <lambda>
    self.Bind(wx.EVT_MENU, lambda event: self.SelectNextItem(True), id=adownId)
  File "/home/laurens/Documents/thesis/tribler/Tribler/Main/vwxGUI/__init__.py", line 113, in invoke_func
    return func(*args, **kwargs)
  File "/home/laurens/Documents/thesis/tribler/Tribler/Main/vwxGUI/list_body.py", line 1304, in SelectNextItem
    select].GetSize()[1] + 1) / self.GetScrollPixelsPerUnit()[1]
KeyError: "\xfb+2z}\x8f\xa5\xbb\xdf\x03\x92\xd0\xef\xf7\xa1{'\x01o\xd6"

### LINKED ISSUE #2061: select all with ctrl-a
feature request: press ctrl-a on downloads view to select all.

### PR DESCRIPTION
**Pull request by devos50**
This PR contains GUI improvements to create a more stable WX3 release. The WX3 GUI is currently stable on OS X, Ubuntu and Windows 10.

In this PR:
-removed a `wx.SafeYield()` statement when data became available to show in the lists. I don't have any idea about the purpose of this statement at this position. After some testing, everything was still stable and the bug in #1947 is gone.
-fixed #2048 by not skipping the click event if it is of type `LeftUp`.
-made the connection label when downloading 2px wider so this should fix #1941.
-various thaws and freezes were killing the GUI on Ubuntu. This is probably due to freeze calls in freeze calls which messed up with the internal freeze count in GTK.
-@lfdversluis encountered an issue where a `keyError` would be thrown when scrolling through the list with the up and down keys. I implemented a workaround for this by adding an additional check. Later @lfdversluis mentioned #1540 which is exactly the issue I fixed here.
-the show columns button is now in the filter headers. This is a more visible and logical place to toggle the columns on list items. Also, the name of the menu item has been changed from `show labels` to `show columns`.
-replaced the Bittorrent icon on a download with icons that indicate the current state of the download (inactive/downloading/stopped/seeding etc).
-replaced the `show grid` button with a grid icon that can be clicked.
-all downloads can now be selected using the Ctrl+a key combination.
-fixed a bug that caused the status bar connection gauge to appear with a black background on Windows.
-disabled some more tooltip occurrences on OS X.
-on OS X, items with a text label in a list would be deselected if the mouse was leaving the text label. The strange thing is that this only happens on OS X and not on other platforms so I guess it's something event-related in wx. I'm ignoring the mouse enter/leave events on `BetterText` labels now.
-the `add external .torrent` dialog is now not closed anymore when a duplicate download is being added.

Fixes #1947, fixes #2048, fixes #1941, fixes #2050, fixes #1704, fixes #875, fixes #1585, fixes #1560, fixes #2061, fixes #1540, fixes #1770.

### CODE REVIEW

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by whirm** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm ready for review

File: Tribler/Main/vwxGUI/widgets.py [hunk 1]
```diff
@@ -235,7 +235,7 @@ def __init__(self,
 
         selectedfont = parent.GetFont()
         selectedfont.SetPointSize(normalfont.GetPointSize() + font_increment)
-        selectedfont.SetUnderlined(True)
+        #selectedfont.SetUnderlined(True)
```
Comment (lfdversluis) [inline Tribler/Main/vwxGUI/widgets.py]: why commenting this out? Should it be deleted or?

File: Tribler/Main/vwxGUI/list_body.py [hunk 2]
```diff
@@ -1018,10 +1028,10 @@ def __SetData(self, highlight=True):
             self.CreateItems(nr_items_to_create=3 * LIST_ITEM_BATCH_SIZE)
 
             # Try to yield
-            try:
-                wx.SafeYield()
-            except:
-                pass
+            #try:
```
Comment (lfdversluis) [inline Tribler/Main/vwxGUI/list_body.py]: why the commented out code?

**Comment by lfdversluis** [comment]
Starting from the bottom and working my way to the top I noticed quite a bit of commented out code, should this be considered deleted?

**Comment by whirm** [comment]
Looks good to me, are you planning on adding tests for the non-covered stuff?

**Comment by whirm** [comment]
Ok, @synctext already told me you aren't doing it.

@devos50 let me know if you plan on removing the commented out code.

**Comment by whirm** [comment]
I've tested it on my machine and it seems to work/look just fine

**Comment by devos50** [comment]
@whirm I would like to keep the commented-out code so I can easily trace it back should we encounter a glitch on any platform due to the removal of the thaws/freezes ;). The code is going to be removed soon anyways.

**Comment by whirm** [comment]
Yes, i thought so. I'll merge then.

**Comment by devos50** [comment]
Thanks for the mail spam :) https://gyazo.com/3f1e6199748b2423f8ecf2b4cad97fc9

**Comment by whirm** [comment]
Well, you guys are DDOSing me with PRs! :D

**Pull request merged**